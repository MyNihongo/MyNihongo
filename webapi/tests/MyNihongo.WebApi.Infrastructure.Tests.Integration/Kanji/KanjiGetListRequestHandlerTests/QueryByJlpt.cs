namespace MyNihongo.WebApi.Infrastructure.Tests.Integration.Kanji.KanjiGetListRequestHandlerTests;

[Collection(CollectionDefinitions.Kanji)]
public sealed class QueryByJlpt : KanjiGetListRequestHandlerTestsBase
{
	public QueryByJlpt(KanjiDatabaseSnapshot snapshot)
		: base(snapshot)
	{
	}

	[Fact]
	public async Task QueryFirstPage()
	{
		var request = new KanjiGetListRequest
		{
			PageSize = 15
		};

		var result = await CreateFixture()
			.Handle(request, CancellationToken.None)
			.ToArrayAsync();

		ObjectApprovals.VerifyJson(result);
	}

	[Fact]
	public async Task QuerySecondPage()
	{
		var request = new KanjiGetListRequest
		{
			PageSize = 15,
			PageIndex = 1
		};

		var result = await CreateFixture()
			.Handle(request, CancellationToken.None)
			.ToArrayAsync();

		ObjectApprovals.VerifyJson(result);
	}

	[Fact]
	public async Task QueryFirstUser()
	{
		var request = new KanjiGetListRequest
		{
			UserId = 1L,
			PageSize = 10
		};

		var result = await CreateFixture()
			.Handle(request, CancellationToken.None)
			.ToArrayAsync();

		ObjectApprovals.VerifyJson(result);
	}

	[Fact]
	public async Task QuerySecondUser()
	{
		var request = new KanjiGetListRequest
		{
			UserId = 2L,
			PageSize = 10
		};

		var result = await CreateFixture()
			.Handle(request, CancellationToken.None)
			.ToArrayAsync();

		ObjectApprovals.VerifyJson(result);
	}

	[Fact]
	public async Task FilterByJlptLevel()
	{
		var request = new KanjiGetListRequest
		{
			JlptLevel = JlptLevel.N3,
			PageSize = 15
		};

		var result = await CreateFixture()
			.Handle(request, CancellationToken.None)
			.ToArrayAsync();

		ObjectApprovals.VerifyJson(result);
	}

	[Fact]
	public async Task FilterUserEntries()
	{
		var request = new KanjiGetListRequest
		{
			Filter = KanjiGetListRequest.Types.Filter.UserData,
			UserId = 1L
		};

		var result = await CreateFixture()
			.Handle(request, CancellationToken.None)
			.ToArrayAsync();

		ObjectApprovals.VerifyJson(result);
	}

	[Fact]
	public async Task FilterUserEntriesAndJlptLevel()
	{
		var request = new KanjiGetListRequest
		{
			JlptLevel = JlptLevel.N5,
			Filter = KanjiGetListRequest.Types.Filter.UserData,
			UserId = 1L
		};

		var result = await CreateFixture()
			.Handle(request, CancellationToken.None)
			.ToArrayAsync();

		ObjectApprovals.VerifyJson(result);
	}

	[Fact]
	public async Task FilterFavourites()
	{
		var request = new KanjiGetListRequest
		{
			Filter = KanjiGetListRequest.Types.Filter.Favourites,
			UserId = 2L
		};

		var result = await CreateFixture()
			.Handle(request, CancellationToken.None)
			.ToArrayAsync();

		ObjectApprovals.VerifyJson(result);
	}

	[Fact]
	public async Task FilterFavouritesAndJlptLevel()
	{
		var request = new KanjiGetListRequest
		{
			JlptLevel = JlptLevel.N4,
			Filter = KanjiGetListRequest.Types.Filter.Favourites,
			UserId = 2L
		};

		var result = await CreateFixture()
			.Handle(request, CancellationToken.None)
			.ToArrayAsync();

		ObjectApprovals.VerifyJson(result);
	}

	[Fact]
	public async Task QueryFrench()
	{
		var request = new KanjiGetListRequest
		{
			JlptLevel = JlptLevel.N4,
			Language = Language.French
		};

		var result = await CreateFixture()
			.Handle(request, CancellationToken.None)
			.ToArrayAsync();

		ObjectApprovals.VerifyJson(result);
	}

	[Fact]
	public async Task QuerySpanish()
	{
		var request = new KanjiGetListRequest
		{
			JlptLevel = JlptLevel.N3,
			Language = Language.Spanish
		};

		var result = await CreateFixture()
			.Handle(request, CancellationToken.None)
			.ToArrayAsync();

		ObjectApprovals.VerifyJson(result);
	}

	[Fact]
	public async Task QueryPortuguese()
	{
		var request = new KanjiGetListRequest
		{
			JlptLevel = JlptLevel.N2,
			Language = Language.Portuguese
		};

		var result = await CreateFixture()
			.Handle(request, CancellationToken.None)
			.ToArrayAsync();

		ObjectApprovals.VerifyJson(result);
	}
}