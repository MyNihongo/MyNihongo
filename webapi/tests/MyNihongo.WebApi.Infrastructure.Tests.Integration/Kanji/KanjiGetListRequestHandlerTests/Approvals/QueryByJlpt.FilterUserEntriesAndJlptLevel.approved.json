[
	{
		"KanjiId": 5,
		"SortingOrder": 1,
		"Character": "飲",
		"JlptLevel": "N5",
		"UserData": {
			"FavouriteRating": 0.0
		},
		"Readings": [
			{
				"ReadingType": "On",
				"MainText": "イン",
				"SecondaryText": "",
				"RomajiText": "iso"
			},
			{
				"ReadingType": "Kun",
				"MainText": "の",
				"SecondaryText": "む",
				"RomajiText": "nomu"
			},
			{
				"ReadingType": "On",
				"MainText": "オン",
				"SecondaryText": "",
				"RomajiText": "oso"
			},
			{
				"ReadingType": "Kun",
				"MainText": "の",
				"SecondaryText": "み",
				"RomajiText": "nomi"
			}
		],
		"Meanings": [
			"drink",
			"smoke",
			"take"
		]
	},
	{
		"KanjiId": 7,
		"SortingOrder": 3,
		"Character": "休",
		"JlptLevel": "N5",
		"UserData": {
			"FavouriteRating": 0.0
		},
		"Readings": [
			{
				"ReadingType": "On",
				"MainText": "キュウ",
				"SecondaryText": "",
				"RomajiText": "kyuu"
			},
			{
				"ReadingType": "Kun",
				"MainText": "やす",
				"SecondaryText": "む",
				"RomajiText": "yasumu"
			},
			{
				"ReadingType": "Kun",
				"MainText": "やす",
				"SecondaryText": "まる",
				"RomajiText": "yasumaru"
			},
			{
				"ReadingType": "Kun",
				"MainText": "やす",
				"SecondaryText": "める",
				"RomajiText": "yasumeru"
			}
		],
		"Meanings": [
			"rest",
			"day off",
			"retire",
			"sleep"
		]
	}
]